homeassistant/components/google/__init__.py

```diff
@@ -117,12 +129,74 @@
     {
         vol.Required(EVENT_CALENDAR_ID): cv.string,
         vol.Required(EVENT_SUMMARY): cv.string,
+        vol.Optional(EVENT_SENDUPDATES, default="none"): vol.In(
+            ["all", "externalOnly", "none"]
+        ),
         vol.Optional(EVENT_DESCRIPTION, default=""): cv.string,
         vol.Exclusive(EVENT_START_DATE, EVENT_START_CONF): cv.date,
         vol.Exclusive(EVENT_END_DATE, EVENT_END_CONF): cv.date,
         vol.Exclusive(EVENT_START_DATETIME, EVENT_START_CONF): cv.datetime,
         vol.Exclusive(EVENT_END_DATETIME, EVENT_END_CONF): cv.datetime,
         vol.Exclusive(EVENT_IN, EVENT_START_CONF, EVENT_END_CONF): _EVENT_IN_TYPES,
+        vol.Optional(EVENT_LOCATION, default=""): cv.string,
+        vol.Optional(EVENT_RECURRENCE, default=[]): cv.ensure_list_csv,
+        vol.Optional(EVENT_STATUS, default="confirmed"): vol.In(
+            ["confirmed", "tentative", "cancelled"]
+        ),
+        vol.Optional(EVENT_TRANSPARENCY): vol.In(["opaque", "transparent"]),
+        vol.Optional(EVENT_VISIBILITY, default="default"): vol.In(
+            ["default", "public", "private", "confidential"]
+        ),
+        vol.Optional(EVENT_REMINDERS_USEDEFAULT, default="false"): cv.boolean,
+        vol.Optional(EVENT_REMINDERS_OVERRIDES_METHOD, default="popup"): vol.In(
+            ["email", "popup"]
+        ),
+        vol.Optional(EVENT_REMINDERS_OVERRIDES_MINUTES, default=10): vol.All(
+            vol.Coerce(int), vol.Range(min=1, max=40320)
+        ),
+    }
+)
+
+DELETE_EVENT_SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(EVENT_CALENDAR_ID): cv.string,
+        vol.Required(EVENT_ID): cv.string,
+        vol.Optional(EVENT_SENDUPDATES, default="none"): vol.In(
+            ["all", "externalOnly", "none"]
+        ),
+    }
+)
+
+UPDATE_EVENT_SERVICE_SCHEMA = vol.Schema(
```

Comment:
It would make more sense if we make this a generic calendar service and allow integrations to specify if they support updating the calendar.